src/signature.cpp

```diff
@@ -0,0 +1,290 @@
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file signature.cpp Implementation of signature validation routines. */
+
+#include "stdafx.h"
+
+#include "signature.h"
+
+#include "debug.h"
+#include "fileio_func.h"
+#include "string_func.h"
+
+#include "3rdparty/monocypher/monocypher.h"
+#include "3rdparty/nlohmann/json.hpp"
+
+#include "safeguards.h"
+
+/** The public keys used for signature validation. */
+static std::vector<std::string> _public_keys_v1 = {
+	"1234567890123456789012345678901234567890123456789012345678901234",
+	"1234567890123456789012345678901234567890123456789012345678901234",
```

Comment:
Ofc these are not valid yet; I need to generate a private/public pair and add the private part to the CI where needed. And the public pair here.